core/src/main/java/org/apache/iceberg/GenericDataFile.java

```diff
@@ -66,23 +68,31 @@ class GenericDataFile extends BaseFile<DataFile> implements DataFile {
    * Copy constructor.
    *
    * @param toCopy a generic data file to copy.
-   * @param fullCopy whether to copy all fields or to drop column-level stats
+   * @param copyStats whether to copy all fields or to drop column-level stats.
+   * @param columnsToKeepStats a set of column ids to keep stats. If empty or <code>null</code> then
+   *     every column stat is kept.
    */
-  private GenericDataFile(GenericDataFile toCopy, boolean fullCopy) {
-    super(toCopy, fullCopy);
+  private GenericDataFile(
+      GenericDataFile toCopy, boolean copyStats, Set<Integer> columnsToKeepStats) {
```

Comment:
I think the object name is confusing. It doesn't need to match the method name and should be more straightforward. How about `requestedColumns` or something?

Reply:
Done